### PR TITLE
[WIP][deploy][rps] models of resource provision service

### PR DESCRIPTION
**Pull request by zeromem**
## Why are these changes needed?

We plan to implement a Resource Provision Service (RPS) for Ray, you can read the design doc [here](https://docs.google.com/document/d/1Isx6I5dF2jRdNh9RVIRVmYb4Rd0QxOlAd_glq_l0eJo).

We have implemented a kubernetes operator for Ray (and has been merged #6328 ). As we know "Desired State Oriented" is an important core concept in Kubernetes, it means we need to interact with Kubernetes using declarative API, so we need a translator to translate user commands(from ray autoscaler or any other user client) to a desired ray cluster state.

On the other side, we can abstract the interaction with Kubernetes and Yarn to make the deployment and management more concise and unified.

In this plan, We will 
1. provide the abstraction / interface between Ray and different platforms like Kubernetes and Yarn. (This is what this Pull Request for).
2. provide a k8s-implementation.
3. adapt the ray-operator for the RPS.

## Related issue number

Merged #6328 
Merged #6501 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: deploy/java/ray-rps-k8s-impl/src/main/java/org/ray/deploy/rps/k8s/model/BaseCustomResource.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package org.ray.deploy.rps.k8s.model;
+
+import io.fabric8.kubernetes.client.CustomResource;
+import java.util.HashMap;
+import java.util.Map;
+
+public class BaseCustomResource extends CustomResource {
+    public static final String LABEL_POOL = "mandatory.k8s.alipay.com/app-logic-pool";
```
Comment (caitengwei) [inline deploy/java/ray-rps-k8s-impl/src/main/java/org/ray/deploy/rps/k8s/model/BaseCustomResource.java]: Is this label necessary?

File: deploy/java/ray-rps-model/src/main/java/org/ray/deploy/rps/model/Request.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Alipay.com Inc. Copyright (c) 2004-2019 All Rights Reserved.
+ */
```
Comment (caitengwei) [inline deploy/java/ray-rps-model/src/main/java/org/ray/deploy/rps/model/Request.java]: Pls remove these lines

**Review by caitengwei — CHANGES_REQUESTED** [review]
@zeromem Can you add a doc for RPS's API?

**Comment by richardliaw** [comment]
OK I'm going to close this because there's no development activity. Others have mentioned to me that this would have been useful, but I think we need to resurface the discussion if we wanted to invest time on this.

**Comment by richardliaw** [comment]
Please open a new issue if you wish to discuss deployment.

**Pull request closed without merging**